### PR TITLE
fix(lxd): Properly handle unicode from LXD socket

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
fix(lxd): Properly handle unicode from LXD socket

Fixes GH-5300
```

## Additional Context
See #5300

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Review by blackboxsw — COMMENTED** [review]
functionally looks good. 
I think we are missing correct mocks here in tests/unittests/sources/test_lxd.py::TestReadMetadata::test_read_metadata_handles_unexpected_content_or_http_status as we are failing on our mock here in py 3.11 due to our mocking of response.text instead of response.content.  May need to update that unittest to avoid the 8 failures there.

**Comment by TheRealFalcon** [comment]
Doh, I missed the unit test failures. Thanks! Should be updated now.

File: tests/integration_tests/modules/test_combined.py [hunk 1]
```diff
@@ -506,6 +507,10 @@ def test_instance_cloud_id_across_reboot(
         assert client.execute(f"test -f /run/cloud-init/{cloud_file}").ok
         assert client.execute("test -f /run/cloud-init/cloud-id").ok
 
+    def test_unicode(self, class_client: IntegrationInstance):
+        client = class_client
+        assert "💩" == client.read_from_file("/var/tmp/unicode_data")
```
Comment (blackboxsw) [inline tests/integration_tests/modules/test_combined.py:512]: And, who says we don't know how to have fun in this project?

**Review by blackboxsw — APPROVED** [review]
Bring it home!

**Pull request merged**